src/protocol.rs

```diff
@@ -313,42 +324,60 @@ impl std::fmt::Display for ParticipantIdentifier {
 
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
 
-/// An [`Identifier`] is a session identifier that uniquely identifies a single
+/// A session [`Identifier`] uniquely identifies a single
 /// instance of a protocol and all messages associated with it.
 ///
-/// [`Identifier`] are globally unique identifiers that allow parties to
-/// distinguish between different sessions. The "globally unique" property
-/// of these identifiers forces parties to avoid any collisions between
-/// different sessions and any kind of replay attack by associating messages,
-/// parameters, proofs and commitments with their corresponding sessions.
+/// Session identifiers have two roles in the protocol: they tag messages
+/// and they are incorporated as context into zero-knowledge proofs.
+/// The "globally unique" property allows participants to distinguish
+/// messages belonging to different, concurrent protocol runs,
+/// prevents collisions between messages belonging to
+/// different sessions, and prevents replay attacks by associating messages and
+/// zero-knowledge proofs
+/// with the session, fixed parameters, and previous subprotocols to which
+/// they correspond.
 ///
 /// # Discrepancies with the paper with respect to session identifiers:
+/// The paper defines session identifiers, denoted `sid` and `ssid`, somewhat
+/// differently that we implement them in this codebase. We believe this
+/// implementation is functionally equivalent to what the paper describes.
+///
+/// 1. The paper incorporates many types of data into its session and
+/// sub-session identifiers, including fixed parameters, the participant set,
+/// and outputs from previous protocols; these identifiers are used both to tag
+/// messages and incorporate context into proofs. The codebase defines a single
+/// `Identifier` type that is expected to be chosen uniformly at random; this is
```

Comment:
this requirement seems to contradict what we say below about options for picking session identifiers

Reply:
Right. I will rephrase this to emphasize the property (global, unique) that these tags have, and not the method of choosing them.